scripts/gitignore_remove_generated_files.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+set -eu
+
+GITIGNORES=$(find . -name ".gitignore")
+
+for GITIGNORE in $GITIGNORES; do
+    IN_GEN_BLOCK=false
+    while read -r line; do
+        if [ "$line" = "###START_GENERATED_FILES###" ]; then
+            IN_GEN_BLOCK=true
+            echo "###START_COMMENTED_GENERATED_FILES###"
+        elif [ "$line" = "###END_GENERATED_FILES###" ]; then
+            IN_GEN_BLOCK=false
+            echo "###END_COMMENTED_GENERATED_FILES###"
+        elif $IN_GEN_BLOCK ; then
+            echo "#$line"
+        else
+            echo "$line"
+        fi
+    done <$GITIGNORE > "$GITIGNORE.tmp"
+    mv "$GITIGNORE.tmp" $GITIGNORE
+done
```

Comment:
I would unify those files to a single simply named file .e.g `gitignore_patch.sh` which is easier to review, and running with command line parameters:

- No input arguments or `-h` `--help` -> Echo a documentation/syntax message
-  `--auto-gen-en`  To enable autogenerated files tracking
-  `--auto-gen-dis`  To disable autogenerated files tracking

Feel free to come up wth more creative names or line arguments.